tests/Compiler/Mapper/Scalar/Data/FloatWithAllowedInfinityAndNanMapper.php

```diff
@@ -26,10 +26,14 @@ public function __construct(private readonly MapperProvider $provider)
      */
     public function map(mixed $data, array $path = []): float
     {
-        if (!is_float($data) && !is_int($data)) {
+        if (is_float($data)) {
+            $mapped = $data;
+        } elseif (is_int($data) && $data >= -9007199254740991 && $data <= 9007199254740991) {
```

Comment:
Probably nitpicking, but since the const in MapFloat is public, we might reuse it here to make it more readable even within the compiled code. Or re-declare the const here.